yt_dlp/extractor/mediastream.py

```diff
@@ -118,7 +122,36 @@ def _real_extract(self, url):
         formats, subtitles = [], {}
         for video_format in player_config['src']:
             if video_format == 'hls':
-                fmts, subs = self._extract_m3u8_formats_and_subtitles(player_config['src'][video_format], video_id)
+                src = player_config['src'][video_format]
+                params = {}
+
+                uid = self._search_regex(r'window\.MDSTRMUID\s*=\s*["\']([^"\']+)["\'];', webpage, 'uid', fatal=False, default=None)
+                if uid:
+                    params['uid'] = uid
+
+                sid = self._search_regex(r'window\.MDSTRMSID\s*=\s*["\']([^"\']+)["\'];', webpage, 'sid', fatal=False, default=None)
+                if sid:
+                    params['sid'] = sid
+
+                pid = self._search_regex(r'window\.MDSTRMPID\s*=\s*["\']([^"\']+)["\'];', webpage, 'pid', fatal=False, default=None)
+                if pid:
+                    params['pid'] = pid
+
+                version = self._search_regex(r'window\.VERSION\s*=\s*["\']([^"\']+)["\'];', webpage, 'version', fatal=False, default=None)
+                if version:
+                    params['at'] = 'web-app'
+                    params['av'] = version
+
+                parsed = urlparse(url)
+                if len(parsed.query) > 0:
+                    qs = parse_qs(parsed.query)
+                    if 'access_token' in qs and len(qs['access_token']) > 0:
+                        params['access_token'] = qs['access_token'][0]
```

Comment:
Since we no longer support 3.7, we could also do:
```py
                if access_token := parse_qs(url).get('access_token', [None])[0]:
                    params['access_token'] = access_token
```